documentation/manual/source/pages/development/using_non_qx_libs.rst

```diff
@@ -66,31 +66,76 @@ There is a more elaborate way to use an external %{JS} package, by wrapping it i
 - Exposing the package's API through a qooxdoo class.
 - Automatic dependency management wherever the wrapper class is used.
 
-Here are the necessary steps.
+Step-by-Step Instructions
+--------------------------
 
-1. Create a fresh %{qooxdoo} project using `create-application.py`.
-2. As before, copy the external package's files to a suitable place under the `resource` folder, e.g. `source/resource/ponycharts/ponycharts.js`
-3. Add an `@asset` hint in the project's main class
+#. Create a fresh %{qooxdoo} project using `create-application.py`.
+#. As in the approach above, copy the external package's files to a suitable place under the `resource` folder, e.g. `source/resource/ponycharts/ponycharts.js`.
+#. Likewise, add an `@asset` hint in the library's main class to include the external package as a resource (full example further below).
+#. Add code to this class that loads the external package, does necessary initialization and potentially adds a %{qooxdoo}-ish API to it.
 
-    ::
+   There is a framework class to help you with the loading part, `qx.util.DynamicScriptLoader <http://demo.qooxdoo.org/%{version}/apiviewer/#qx.util.DynamicScriptLoader>`_, which does most of the work to make the package available in the current browser context. Using this, here is how your wrapper class may look like:
+
+   ::
 
         /**
          * This is the main class of the PonyCharts wrapper.
          *
-         * @asset(ponycharts/ponycharts.js)
+         * @asset(ponycharts/*)
          */
         qx.Class.define("ponycharts.PonyCharts",
-            ...
+        {
+          extend: qx.core.Object,
+
+          construct: function() {
+            // initialize the script loading
+            var dynLoader = new qx.util.DynamicScriptLoader([
+                "ponycharts/PonyCharts.js"
+            ]);
+
+            dynLoader.addListenerOnce('ready',function(e){
+              console.log("all scripts have been loaded!");
```

Comment:
A code fragment or a call to something like ```initializePonyCharts``` could be placed here to show the reader that this is the right place to do some post loading tasks.